authentification/usecase/login.go

```diff
@@ -0,0 +1,49 @@
+package usecase
+
+import (
+	"context"
+	"log"
+	"os"
+
+	"github.com/There-is-Go-alternative/GoMicroServices/authentification/database"
+	"github.com/There-is-Go-alternative/GoMicroServices/authentification/domain"
+	"go.mongodb.org/mongo-driver/bson"
+)
+
+type LoginDTO struct {
+	Email    string `json:"email"`
+	Password string `json:"password"`
+}
+
+type LoginProto func(ctx context.Context, input LoginDTO) (*domain.Token, error)
+
+func Login() LoginProto {
+	return func(ctx context.Context, input LoginDTO) (*domain.Token, error) {
+		collection, err := database.GetMongoDbCollection(os.Getenv("MONGO_DB"), os.Getenv("MONGO_COLLECTION"))
+		if err != nil {
+			return nil, err
+		}
+
+		var auth domain.Auth
+		err = collection.FindOne(ctx, bson.M{"email": input.Email}).Decode(&auth)
+		if err != nil {
+			log.Println(err)
+			return nil, err
+		}
```

Comment:
Tu n'abstrait pas ta DB, si je veux utiliser une autre Databse, je dois aller fouiller dans ton code pour retirer tout ce qui est en rapport avec MongoDB.

database.GetMongoDbCollection est une fonction qui n'est rien à faire dans ton usecase.
Pareil pour FindOne, qui prend un bson.M, spécifique à MongoDB

tu devrais avoir ici:
```suggestion
type database interface {
   FindOne(context.Context, string) (domain.Auth, error)
}

func Login(db database) LoginProto {
	return func(ctx context.Context, input LoginDTO) (*domain.Token, error) {
	        auth, err := db.FindOne(ctx, input.Email)
		if err != nil {
			return nil, err
		}
		
		....
		
	}
}
```

Et dans ton fichier relatif à mongodb:

```
type MogoDB struct {
    // ton pointeur vers ta DB ici
}

func (db MongoDB) FindOne(ctx context.Context, email string) (domain.Auth, error) {
		var auth domain.Auth
		err = collection.FindOne(ctx, bson.M{"email": email}).Decode(&auth)
		if err != nil {
			log.Println(err)
			return domain.Auth{}, err
		}
		return auth, nil
}
```